lib/spack/docs/packaging_guide.rst

```diff
@@ -5388,6 +5388,86 @@ follows:
 
 .. _package-lifecycle:
 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Add detection tests to packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To ensure that a software is detected correctly for multiple configurations
+and on different systems users can write a ``detection_test.yaml`` file and
+put it in the same directory as the corresponding ``package.py`` file.
+This YAML file contains enough information for Spack to mock an environment
+and try to check if the detection logic yields the results that are expected.
+
+As a general rule, attributes at the top-level of ``detection_test.yaml``
+represent search mechanisms and they all map to a list of tests that should confirm
```

Comment:
```suggestion
represent search mechanisms and they each map to a list of tests that should confirm
```

Reply:
Done in https://github.com/spack/spack/pull/18175/commits/d1c916b50af4517a2ee0b09f794fabceaedf3e2d